File: dvbViewerController/src/main/java/org/dvbviewer/controller/io/api/DMSInterface.java
package org.dvbviewer.controller.io.api;

import org.dvbviewer.controller.data.media.xml.VideoDirsFiles;
import org.dvbviewer.controller.data.task.xml.TaskList;
import org.dvbviewer.controller.data.version.xml.Version;

import java.util.Map;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.QueryMap;

public interface DMSInterface {

    String API = "api/";

    String TASK_API                 = API + "tasks.html";

    String REC_DEL_API              = API + "recdelete.html";

    String TIMER_ADD_API            = API + "timeradd.html";

    String TIMER_EDIT_API           = API + "timeredit.html";

    String MEDIA_DIRS               = API + "mediafiles.html?content=3&recursive=0&thumbs=1";

    String VERSION               = API + "version.html";

    @GET(TASK_API)
    Call<TaskList> getTaskList(@Query("all") int all);

    @GET(TASK_API)
    Call<ResponseBody> executeTask(@Query("action") String action);

    @GET(REC_DEL_API)
    Call<ResponseBody> deleteRecording(@Query("recid") long id, @Query("delfile") int delete);

    @GET(TIMER_ADD_API)
    Call<ResponseBody> addTimer(@QueryMap Map<String, String> params);

    @GET(TIMER_EDIT_API)
    Call<ResponseBody> editTimer(@QueryMap Map<String, String> params);

    @GET(MEDIA_DIRS)
    Call<VideoDirsFiles> getMediaDir(@Query("dirid") long id);

    @GET(VERSION)
    Call<Version> getVersion();

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/io/api/DMSInterface.kt
package org.dvbviewer.controller.io.api

import okhttp3.ResponseBody
import org.dvbviewer.controller.data.media.xml.VideoDirsFiles
import org.dvbviewer.controller.data.task.xml.TaskList
import org.dvbviewer.controller.data.version.xml.Version
import org.dvbviewer.controller.entities.FFMpegPresetList
import org.dvbviewer.controller.entities.Timer
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.QueryMap

interface DMSInterface {

    @get:GET(VERSION)
    val version: Call<Version>


    @GET(TASK_API)
    fun getTaskList(@Query("all") all: Int): Call<TaskList>

    @GET(TASK_API)
    fun executeTask(@Query("action") action: String): Call<ResponseBody>

    @GET(REC_DEL_API)
    fun deleteRecording(@Query("recid") id: Long, @Query("delfile") delete: Int): Call<ResponseBody>

    @GET(TIMER_LIST)
    fun getTimer(): Call<List<Timer>>

    @GET(TIMER_ADD_API)
    fun addTimer(@QueryMap params: Map<String, String>): Call<ResponseBody>

    @GET(TIMER_DELETE)
    fun deleteTimer(@Query("id") id: String): Call<ResponseBody>

    @GET(MEDIA_DIRS)
    fun getMediaDir(@Query("dirid") id: Long): Call<VideoDirsFiles>

    @GET(CONFIG_FILE)
    fun getConfigFile(@Query("file") file: String): Call<FFMpegPresetList>

    companion object {

        const val API = "/api"

        const val TASK_API = "$API/tasks.html"

        const val REC_DEL_API = "$API/recdelete.html"

        const val TIMER_LIST = "$API/timerlist.html?utf8=2"

        const val TIMER_ADD_API = "$API/timeradd.html"

        const val TIMER_DELETE = "$API/timerdelete.html"

        const val MEDIA_DIRS = "$API/mediafiles.html?content=3&recursive=0&thumbs=1"

        const val VERSION = "$API/version.html"

        const val CONFIG_FILE = "$API/api/getconfigfile.html"
    }

}